livekit-agents/livekit/agents/pipeline/metrics.py

```diff
@@ -0,0 +1,52 @@
+from typing import Literal, TypedDict
+
+from livekit.rtc import EventEmitter
+
+MetricsEventTypes = Literal[
+    "vad_metrics_collected",
+    "stt_metrics_collected",
+    "llm_metrics_collected",
+    "tts_metrics_collected",
+]
+
+
+class PipelineMetrics(EventEmitter[MetricsEventTypes]): ...
+
+
+# VAD Metrics are currently being sent every 1s
+class VADMetrics(TypedDict):
+    type: Literal["vad_metrics"]
+    timestamp: float
+    avg_inference_duration: float
+    inference_count: int
+
+
+class STTMetrics(TypedDict):
+    type: Literal["stt_metrics"]
+    timestamp: float
+    speech_id: str
+    estimated_ttfb: float
+    """
+    The estimated time-to-first-byte (TTFB) of the STT service.
+    This is calculated using the VAD provided inside the PipelineAgent.
+    """
+
+
+class LLMMetrics(TypedDict):
```

Comment:
can we also have token usage metrics in here? along with the input and output of that inference?

Reply:
future PR